src/sys/statfs.rs

```diff
@@ -1,22 +1,432 @@
+        self.0.f_bsize
+    }
+
+    /// Optimal transfer block size
+    #[cfg(any(
+        target_os = "android",
+        all(
+            target_os = "linux",
+            target_env = "musl"
+        )
+    ))]
+    pub fn optimal_transfer_size(&self) -> libc::c_ulong {
+        self.0.f_bsize
+    }
+
+    /// Optimal transfer block size
+    #[cfg(all(
+        target_os = "linux",
+        not(target_env = "musl"),
+        not(target_arch = "s390x")
+    ))]
+    pub fn optimal_transfer_size(&self) -> libc::c_long {
+        self.0.f_bsize
+    }
+
+    /// Optimal transfer block size
+    #[cfg(target_os = "freebsd")]
+    pub fn optimal_transfer_size(&self) -> libc::uint64_t {
+        self.0.f_iosize
+    }
+
+    /// Size of a block
+    #[cfg(any(
+        target_os = "ios",
+        target_os = "macos"
+    ))]
+    pub fn block_size(&self) -> libc::uint32_t {
+        self.0.f_bsize
+    }
+
+    /// Size of a block
+    #[cfg(all(
+        target_os = "linux",
+        target_arch = "s390x"
+    ))]
+    pub fn block_size(&self) -> libc::uint32_t {
+        self.0.f_frsize
+    }
+
+    /// Size of a block
+    #[cfg(target_os = "freebsd")]
+    pub fn block_size(&self) -> libc::uint64_t {
+        self.0.f_bsize
+    }
+
+    /// Size of a block
+    #[cfg(any(
+        target_os = "android",
+        all(
+            target_os = "linux",
+            target_env = "musl"
+        )
+    ))]
+    pub fn block_size(&self) -> libc::c_ulong {
+        self.0.f_frsize
+    }
+
+    /// Size of a block
+    #[cfg(all(
+        target_os = "linux",
+        not(any(
+            target_arch = "s390x",
+            target_env = "musl"
+        ))
+    ))]
+    pub fn block_size(&self) -> libc::c_long {
+        self.0.f_frsize
+    }
+
+    /// Maximum length of filenames
+    #[cfg(any(
+        target_os = "android",
+        all(
+            target_os = "linux",
+            target_env = "musl"
+        )
+    ))]
+    pub fn maximum_name_length(&self) -> libc::c_ulong {
+        self.0.f_namelen
+    }
+
+    /// Maximum length of filenames
+    #[cfg(all(
+        target_os = "linux",
+        target_arch = "s390x"
+    ))]
+    pub fn maximum_name_length(&self) -> libc::uint32_t {
+        self.0.f_namelen
+    }
+
+    /// Maximum length of filenames
+    #[cfg(target_os = "freebsd")]
+    pub fn maximum_name_length(&self) -> libc::uint32_t {
+        self.0.f_namemax
+    }
 
-        Errno::result(res).map(drop)
+    /// Maximum length of filenames
+    #[cfg(all(
+        target_os = "linux",
+        not(target_arch = "s390x"),
+        not(target_env = "musl")
+    ))]
+    pub fn maximum_name_length(&self) -> libc::c_long {
+        self.0.f_namelen
+    }
+
+    /// Total data blocks in filesystem
+    #[cfg(any(
+        target_os = "ios",
+        target_os = "macos",
+        target_os = "android",
+        target_os = "freebsd",
+        target_pointer_width = "64",
```

Comment:
Here and elsewhere, this condition is confusing and incorrect.  Better written as `all(target_os = "linux", target_pointer_width = "64")`.  OpenBSD should be added to the list here too.